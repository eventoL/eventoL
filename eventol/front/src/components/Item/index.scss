@import '../../styles/variables';

.item {
  -ms-flex: 1 0 auto;
  -webkit-box-flex: 1;
  -webkit-flex: 1 0 auto;
  -webkit-transition: background 1s ease;
  background-clip: content-box;
  background-color: $white;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  background-size: 100%;
  cursor: pointer;
  flex: 1 0 auto;
  height: 200px;
  margin-right: 30px;
  overflow: hidden;
  transition: background 1s ease;

  a {
    text-decoration: none;
  }

  .overlay {
    -webkit-transition: opacity .125s ease;
    // FF3.6-15
    background: -webkit-linear-gradient(top, $gradient-color 0%, transparent 100%);
    // Chrome10-25,Safari5.1-6
    background: linear-gradient(to bottom, $gradient-color 0%, transparent 100%);
    // W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+
    color: $light-gray;
    filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#a6000000', endColorstr='#00000000',GradientType=0 );
    height: 100%;
    opacity: 0;
    padding: 20px;
    pointer-events: none;
    position: relative;
    transition: opacity .125s ease;

    .title {
      -webkit-transition: opacity .25s ease;
      font-size: 22px;
      font-weight: 500;
      opacity: 0;
      transition: opacity .25s ease;
    }

    .rating {
      -webkit-transition: opacity .25s ease .125s;
      font-size: 14px;
      opacity: 0;
      transition: opacity .25s ease .125s;

      &.tags {
        div {
          border-radius: 5px;
          display: inline;
          padding: 2px;

          &:hover {
            background-color: $eventol-color;
          }
        }
      }
    }

    .plot {
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 3;
      -webkit-transition: margin .25s ease .125s, opacity .25s ease .25s;
      display: -webkit-box;
      font-size: 14px;
      font-weight: 500;
      line-height: 1.6;
      margin-top: 100px;
      opacity: 0;
      overflow: hidden;
      text-overflow: ellipsis;
      transition: margin .25s ease .125s, opacity .25s ease .25s;
    }
  }
}

.item:hover {
  background-size: 150%;

  .overlay {
    opacity: 1;
    pointer-events: all;

    .plot {
      margin-top: 10px;
      opacity: 1;
    }

    .title {
      opacity: 1;
    }

    .rating {
      opacity: 1;
    }

    .list-toggle {
      opacity: 1;
    }
  }
}
