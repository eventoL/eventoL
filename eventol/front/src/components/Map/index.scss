@import '../../styles/variables';

.leaflet-container {
  font: unset;

  a {
    color: unset;
  }
}

.item {
  .overlay {
    z-index: 500;
  }

  .max-size {
    max-width: 22vw;
  }
}
