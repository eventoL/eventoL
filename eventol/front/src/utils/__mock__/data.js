const place1 = {
  address_components: [
    {long_name: '1667', short_name: '1667', types: ['street_number']},
    {long_name: 'Ayacucho', short_name: 'Ayacucho', types: ['route']},
    {
      long_name: 'Centro',
      short_name: 'Centro',
      types: ['neighborhood', 'political'],
    },
    {
      long_name: 'Rosario',
      short_name: 'Rosario',
      types: ['locality', 'political'],
    },
    {
      long_name: 'Departamento Rosario',
      short_name: 'Departamento Rosario',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Santa Fe',
      short_name: 'Santa Fe',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'S2000', short_name: 'S2000', types: ['postal_code']},
    {long_name: 'BYS', short_name: 'BYS', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Ayacucho 1667, S2000BYS Rosario, Santa Fe, Argentina',
  geometry: {
    location: {lat: -32.9586923, lng: -60.629729},
    location_type: 'ROOFTOP',
    viewport: {
      south: -32.9600412802915,
      west: -60.63107798029148,
      north: -32.9573433197085,
      east: -60.628380019708516,
    },
  },
  place_id: 'ChIJ7QvrMwCrt5URpyBml6doLnY',
  plus_code: {
    compound_code: '29RC+G4 Rosario, Santa Fe, Argentina',
    global_code: '47VX29RC+G4',
  },
  types: ['street_address'],
};

const place2 = {
  address_components: [
    {long_name: '1551', short_name: '1551', types: ['street_number']},
    {long_name: 'Sarmiento', short_name: 'Sarmiento', types: ['route']},
    {
      long_name: 'San Nicolás',
      short_name: 'San Nicolás',
      types: ['political', 'sublocality', 'sublocality_level_1'],
    },
    {
      long_name: 'Comuna 1',
      short_name: 'Comuna 1',
      types: ['administrative_area_level_2', 'political'],
    },
    {
      long_name: 'Buenos Aires',
      short_name: 'CABA',
      types: ['administrative_area_level_1', 'political'],
    },
    {long_name: 'Argentina', short_name: 'AR', types: ['country', 'political']},
    {long_name: 'C1042', short_name: 'C1042', types: ['postal_code']},
    {long_name: 'ABC', short_name: 'ABC', types: ['postal_code_suffix']},
  ],
  formatted_address: 'Sarmiento 1551, C1042ABC CABA, Argentina',
  geometry: {
    location: {lat: -34.6052436, lng: -58.388569700000005},
    location_type: 'ROOFTOP',
    viewport: {
      south: -34.6065925802915,
      west: -58.38991868029149,
      north: -34.60389461970851,
      east: -58.387220719708466,
    },
  },
  place_id: 'ChIJbXpzCcTKvJURbeIVAyh6SnU',
  plus_code: {
    compound_code: '9JV6+WH Buenos Aires, Argentina',
    global_code: '48Q39JV6+WH',
  },
  types: ['street_address'],
};

export const event1 = {
  attendees: 10,
  backdrop: 'event1backdrop',
  eventSlug: 'event1slug',
  overview: 'event1overview',
  place: JSON.stringify(place1),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 1',
  url: 'event1Url',
  id: 1,
};

export const event2 = {
  attendees: 25,
  backdrop: 'event2backdrop',
  eventSlug: 'event2slug',
  overview: 'event2overview',
  place: JSON.stringify(place2),
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 2',
  url: 'event2Url',
  id: 2,
};

export const events = [event1, event2];

export default {
  event1,
  event2,
  events,
};
