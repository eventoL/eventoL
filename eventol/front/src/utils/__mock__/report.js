const getConfirmationReport = (confirmed, notConfirmed) => ({
  confirmed,
  not_confirmed: notConfirmed,
  total: confirmed + notConfirmed,
});

export const usersReports = {
  attendee: {
    with_event_user: getConfirmationReport(1, 2),
    without_event_user: getConfirmationReport(2, 3),
  },
  collaborator: getConfirmationReport(3, 4),
  event_user: getConfirmationReport(4, 5),
  installer: getConfirmationReport(6, 7),
  organizer: getConfirmationReport(7, 8),
  speakers: 9,
};

export const activitiesReport = {
  level_count: {
    '1': 3,
    '2': 2,
    '3': 1,
  },
  status_count: {
    '1': 2,
    '2': 1,
  },
  type_count: {
    Talk: 4,
    Workshop: 2,
  },
  ...getConfirmationReport(1, 2),
};

export const installationsReport = {
  hardware_count: {
    notebook: 1,
    phone: 2,
  },
  software_count: {
    ubuntu: 3,
    fedora: 2,
    arch: 1,
  },
  installer_count: {
    '1': 2,
    '2': 1,
  },
  total: 2,
};

export const eventReport = {
  // users data
  ...usersReports,

  // activities data
  activity: activitiesReport,

  // installations data
  installation: installationsReport,
};

export const event1PrivateData = {
  id: 1,
  email: 'organizer1@mailinator.com',
  organizers: 'FirstName1 LastName1 Username1',
};

export const event2PrivateData = {
  id: 2,
  email: 'organizer2@mailinator.com',
  organizers: 'FirstName2 LastName2 Username2',
};

export const eventsPrivateData = [
  {...event1PrivateData},
  {...event2PrivateData},
];

export const getEventData = (id, name, tags, report, eventSlug, location) => ({
  id,
  name,
  tags,
  report,
  location,
  event_slug: eventSlug,
});

export const event1Data = getEventData(
  1,
  'event name',
  {name: 'tag name', slug: 'tag slug'},
  eventReport,
  'event_slug',
  ['Capital Federal', 'Pompeya', 'Buenos Aires', 'Argentina']
);

export const event2Data = getEventData(
  2,
  'event 2 name',
  {name: 'tag 2 name', slug: 'tag 2 slug'},
  eventReport,
  'event_2_slug',
  ['Capital Federal', 'Monserrat', 'Buenos Aires', 'Argentina']
);

export const eventsData = [{...event1Data}, {...event2Data}];

export const expectedActivityDetail = {
  levels: {'1': 6, '2': 4, '3': 2},
  status: {'1': 4, '2': 2},
  types: {Talk: 8, Workshop: 4},
};

export const expectedInstallationsSoftwares = {arch: 2, fedora: 4, ubuntu: 6};

export const expectedTotals = {
  activities: {
    confirmed: 2,
    details: expectedActivityDetail,
    not_confirmed: 4,
    total: 6,
  },
  attendees: {confirmed: 6, not_confirmed: 10, total: 16},
  collaborators: {confirmed: 6, not_confirmed: 8, total: 14},
  installations: {softwares: expectedInstallationsSoftwares, total: 4},
  installers: {confirmed: 12, not_confirmed: 14, total: 26},
  organizers: {confirmed: 14, not_confirmed: 16, total: 30},
  speakers: 18,
};

export const expectedEventDataWithoutPrivateData = {
  ...event1Data,
  assistanceDetail: {attendees: {confirmed: 3, not_confirmed: 5, total: 8}},
  locationDetail: {
    address: 'Pompeya',
    address_detail: 'Capital Federal',
    province: 'Buenos Aires',
  },
};

export const expectedEventDataWithPrivateData = {
  ...expectedEventDataWithoutPrivateData,
  ...event1PrivateData,
};

export const expectedValues = {
  expectedTotals,
  expectedActivityDetail,
  expectedInstallationsSoftwares,
  expectedEventDataWithPrivateData,
  expectedEventDataWithoutPrivateData,
};

const mockedData = {
  activitiesReport,
  eventData: event1Data,
  event1Data,
  event2Data,
  eventsData,
  eventReport,
  event1PrivateData,
  event2PrivateData,
  eventsPrivateData,
  getEventData,
  installationsReport,
  usersReports,
  expectedValues,
};
export default mockedData;
