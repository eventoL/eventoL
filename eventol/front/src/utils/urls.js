import {HOME_REQUIRED_FIELDS} from './constants';


// eslint-disable-next-line prefer-destructuring
export const host = window.location.host;

export const EVENTOL_DOCUMENTATION = 'http://eventol.github.io/eventoL/#/';

export const INDEX_URL = '/';
export const REPORT_URL = '/report';
export const LOGIN_URL = '/accounts/login/';
export const LOGOUT_URL = '/accounts/signup/';
export const PROFILE_URL = '/accounts/profile/';

export const API_URL = '/events/api/';
export const EVENTS_API_URL = `${API_URL}events/`;

export const WS_URL = `ws://${host}/`;
export const EVENTS_WS_URL = `${WS_URL}updates/events/`;

export const getTagUrl = slug => `/tags/${slug}`;
export const getEventUrl = slug => `/events/${slug}/`;
export const getApiFullUrl = url => `/api/events/${url}`;
export const getWsUrl = (protocol = 'ws') => `${protocol}://${host}/updates/`;

const addSlugParams = slug => (slug ? `&tags__slug=${slug}` : '');
export const getMyEventsUrl = slug => `?my_events=true&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getSearchUrl = (searchTerm, slug) => `?search=${searchTerm}&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getRecentEventsUrl = slug => `?ordering=-created_at&registration_is_open=true&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getUpcommingEventsUrl = slug => `?registration_is_open=true&ordering=last_date&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getFeaturedEventsUrl = slug => `?ordering=-attendees_count&registration_is_open=true&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getFinishedEventsUrl = slug => `?registration_is_open=false&ordering=-attendees_count&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;
export const getEventsWithConfirmedScheduleUrl = slug => `?schedule_confirmed=true&registration_is_open=true&fields=${HOME_REQUIRED_FIELDS}${addSlugParams(slug)}`;

export const goToUrl = url => {
  window.location.href = url;
};
