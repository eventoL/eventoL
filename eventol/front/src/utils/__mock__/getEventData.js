import {eventReport} from './report';

export const getEventData = (id, name, tags, report, eventSlug, location) => ({
  id,
  name,
  tags,
  report,
  location,
  event_slug: eventSlug,
});
export const event1Data = getEventData(
  1,
  'event name',
  {name: 'tag name', slug: 'tag slug'},
  eventReport,
  'event_slug',
  ['Capital Federal', 'Pompeya', 'Buenos Aires', 'Argentina']
);
export const event2Data = getEventData(
  2,
  'event 2 name',
  {name: 'tag 2 name', slug: 'tag 2 slug'},
  eventReport,
  'event_2_slug',
  ['Capital Federal', 'Monserrat', 'Buenos Aires', 'Argentina']
);
export const eventsData = [{...event1Data}, {...event2Data}];
