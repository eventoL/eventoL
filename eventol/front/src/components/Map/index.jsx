import React from 'react';
import PropTypes from 'prop-types';

import Logger from '../../utils/logger';
import {MAP_SETTINGS, MAP_LAYER} from '../../utils/constants';

import './index.scss';

export default class Map extends React.PureComponent {
  static propTypes = {
    children: PropTypes.element.isRequired,
    eventSlug: PropTypes.string.isRequired,
    place: PropTypes.string.isRequired,
    sliderId: PropTypes.string.isRequired,
  };

  componentDidMount() {
    const eventSlug = this.getMapId();
    try {
      this.loadMap();
    } catch (e) {
      Logger.error(e);
    }
    document.getElementById(eventSlug).classList.remove('max-size');
  }

  getMapId() {
    const {eventSlug, sliderId} = this.props;
    return `${sliderId}${eventSlug}`;
  }

  loadMap() {
    const {place} = this.props;
    const {geometry} = JSON.parse(place);
    const eventLat = geometry.location.lat;
    const eventLon = geometry.location.lng;
    const mapId = this.getMapId();
    const map = L.map(mapId, MAP_SETTINGS).setView([eventLat, eventLon], 5);
    L.tileLayer(MAP_LAYER, {attribution: ''}).addTo(map);
    map.attributionControl.setPrefix('');
    L.marker([eventLat, eventLon]).addTo(map);
  }

  render() {
    const {children} = this.props;
    const mapId = this.getMapId();
    return (
      <div className="item max-size" id={mapId}>
        {children}
      </div>
    );
  }
}
