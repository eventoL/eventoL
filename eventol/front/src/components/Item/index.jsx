import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

import Map from '../Map';
import {getTagUrl, goToUrl} from '../../utils/urls';

import './index.scss';

export default class Item extends React.PureComponent {
  static propTypes = {
    data: PropTypes.shape({
      attendees: PropTypes.number,
      backdrop: PropTypes.string,
      empty: PropTypes.bool,
      eventSlug: PropTypes.string.isRequired,
      overview: PropTypes.string.isRequired,
      place: PropTypes.string.isRequired,
      tags: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          slug: PropTypes.string.isRequired,
        })
      ),
      title: PropTypes.string.isRequired,
      url: PropTypes.string,
    }),
    sliderId: PropTypes.string.isRequired,
  };

  static defaultProps = {
    data: {
      backdrop: null,
      tags: [],
      empty: true,
      url: null,
    },
  };

  getTagLink = ({name, slug}) => (
    <a key={name.toLocaleLowerCase()} href={`${getTagUrl(slug)}`}>
      {name}
    </a>
  );

  handleOnClick = event => {
    event.preventDefault();
    event.stopPropagation();
    const {
      data: {url},
    } = this.props;
    if (url) {
      goToUrl(url);
    }
  };

  getItem = () => {
    const {
      data: {title, attendees, overview, tags, empty},
    } = this.props;
    return (
      <div
        onClick={this.handleOnClick}
        onKeyPress={this.handleOnClick}
        role="button"
        tabIndex="0"
      >
        <div className="overlay">
          <div className="title">{title}</div>
          {!empty && (
            <div className="rating">
              {`${gettext('Attendees')}: ${attendees || 0}`}
            </div>
          )}
          {!_.isEmpty(tags) && (
            <div className="rating tags">
              {/* eslint-disable-next-line react/jsx-one-expression-per-line */}
              {gettext('Tags')}: {tags.map(this.getTagLink)}
            </div>
          )}
          <div className="plot">{overview}</div>
        </div>
      </div>
    );
  };

  render() {
    const {
      data: {backdrop, eventSlug, place},
      sliderId,
    } = this.props;
    const item = this.getItem();
    if (!backdrop)
      return (
        <Map eventSlug={eventSlug} place={place} sliderId={sliderId}>
          {item}
        </Map>
      );
    return (
      <div className="item" style={{backgroundImage: `url(${backdrop})`}}>
        {item}
      </div>
    );
  }
}
