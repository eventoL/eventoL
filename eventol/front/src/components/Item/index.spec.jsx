import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../Map', () => 'Map');

jest.mock('../../utils/urls', () => ({
  getTagUrl: jest.fn().mockImplementation(() => 'tag-url'),
  goToUrl: jest.fn(),
}));

import Map from '../Map';

import {goToUrl} from '../../utils/urls';
import {event1} from '../../utils/__mock__/data';

import Item from '.';

describe('Item', () => {
  let tree;
  let data;
  let event;
  let instance;
  let sliderId;
  let component;

  const renderItem = () => {
    component = renderer.create(<Item data={data} sliderId={sliderId} />);
    tree = component.toJSON();
    instance = component.root;
  };

  beforeEach(() => {
    data = undefined;
    sliderId = event1.id;
    event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
    };
  });

  afterEach(() => {
    goToUrl.mockClear();
  });

  test('Default render', () => {
    renderItem();
    expect(tree).toMatchSnapshot();
  });

  test('should call goToUrl on click event', () => {
    data = event1;
    renderItem();

    expect(goToUrl).not.toBeCalled();

    instance.children[0].props.children.props.onClick(event);
    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();

    expect(goToUrl).toBeCalled();
    expect(goToUrl).toBeCalledWith(data.url);
  });

  test('should call goToUrl on key press event', () => {
    data = event1;
    renderItem();

    expect(goToUrl).not.toBeCalled();

    instance.children[0].props.children.props.onKeyPress(event);

    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();

    expect(goToUrl).toBeCalled();
    expect(goToUrl).toBeCalledWith(data.url);
  });

  test('should shows Map when data has not backdrop', () => {
    data = {...event1, backdrop: undefined};
    renderItem();
    expect(tree).toMatchSnapshot();

    const itemMap = component.root.findByType(Map);
    expect(itemMap).toBeDefined();

    expect(itemMap.props.sliderId).toEqual(sliderId);
    const props = ['eventSlug', 'place'];
    props.forEach(prop => expect(itemMap.props[prop]).toEqual(data[prop]));
  });

  test('Render event 1', () => {
    data = event1;
    renderItem();
    expect(tree).toMatchSnapshot();
  });
});
