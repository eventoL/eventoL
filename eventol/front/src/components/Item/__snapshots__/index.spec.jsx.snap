// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Default render 1`] = `
<Map
  sliderId={1}
>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="overlay"
    >
      <div
        className="title"
      />
      <div
        className="plot"
      />
    </div>
  </div>
</Map>
`;

exports[`Item Render event 1 1`] = `
<div
  className="item"
  style={
    Object {
      "backgroundImage": "url(event1backdrop)",
    }
  }
>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="overlay"
    >
      <div
        className="title"
      >
        event 1
      </div>
      <div
        className="rating"
      >
        Attendees: 10
      </div>
      <div
        className="rating tags"
      >
        Tags
        : 
        <a
          href="tag-url"
        >
          slug-name
        </a>
      </div>
      <div
        className="plot"
      >
        event1overview
      </div>
    </div>
  </div>
</div>
`;

exports[`Item should shows Map when data has not backdrop 1`] = `
<Map
  eventSlug="event1slug"
  place="{\\"address_components\\":[{\\"long_name\\":\\"1667\\",\\"short_name\\":\\"1667\\",\\"types\\":[\\"street_number\\"]},{\\"long_name\\":\\"Ayacucho\\",\\"short_name\\":\\"Ayacucho\\",\\"types\\":[\\"route\\"]},{\\"long_name\\":\\"Centro\\",\\"short_name\\":\\"Centro\\",\\"types\\":[\\"neighborhood\\",\\"political\\"]},{\\"long_name\\":\\"Rosario\\",\\"short_name\\":\\"Rosario\\",\\"types\\":[\\"locality\\",\\"political\\"]},{\\"long_name\\":\\"Departamento Rosario\\",\\"short_name\\":\\"Departamento Rosario\\",\\"types\\":[\\"administrative_area_level_2\\",\\"political\\"]},{\\"long_name\\":\\"Santa Fe\\",\\"short_name\\":\\"Santa Fe\\",\\"types\\":[\\"administrative_area_level_1\\",\\"political\\"]},{\\"long_name\\":\\"Argentina\\",\\"short_name\\":\\"AR\\",\\"types\\":[\\"country\\",\\"political\\"]},{\\"long_name\\":\\"S2000\\",\\"short_name\\":\\"S2000\\",\\"types\\":[\\"postal_code\\"]},{\\"long_name\\":\\"BYS\\",\\"short_name\\":\\"BYS\\",\\"types\\":[\\"postal_code_suffix\\"]}],\\"formatted_address\\":\\"Ayacucho 1667, S2000BYS Rosario, Santa Fe, Argentina\\",\\"geometry\\":{\\"location\\":{\\"lat\\":-32.9586923,\\"lng\\":-60.629729},\\"location_type\\":\\"ROOFTOP\\",\\"viewport\\":{\\"south\\":-32.9600412802915,\\"west\\":-60.63107798029148,\\"north\\":-32.9573433197085,\\"east\\":-60.628380019708516}},\\"place_id\\":\\"ChIJ7QvrMwCrt5URpyBml6doLnY\\",\\"plus_code\\":{\\"compound_code\\":\\"29RC+G4 Rosario, Santa Fe, Argentina\\",\\"global_code\\":\\"47VX29RC+G4\\"},\\"types\\":[\\"street_address\\"]}"
  sliderId={1}
>
  <div
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="overlay"
    >
      <div
        className="title"
      >
        event 1
      </div>
      <div
        className="rating"
      >
        Attendees: 10
      </div>
      <div
        className="rating tags"
      >
        Tags
        : 
        <a
          href="tag-url"
        >
          slug-name
        </a>
      </div>
      <div
        className="plot"
      >
        event1overview
      </div>
    </div>
  </div>
</Map>
`;
